File: library/src/main/java/com/pengrad/telegrambot/request/CopyMessages.java
package com.pengrad.telegrambot.request;

import com.pengrad.telegrambot.response.MessageIdsResponse;


public class CopyMessages extends BaseRequest<CopyMessages, MessageIdsResponse> {

    public CopyMessages(Object chatId, Object fromChatId, int[] messageIds) {
        super(MessageIdsResponse.class);
        add("chat_id", chatId).add("from_chat_id", fromChatId).add("message_ids", messageIds);
    }

    public CopyMessages messageThreadId(Integer messageThreadId) {
        return add("message_thread_id", messageThreadId);
    }

    public CopyMessages removeCaption(boolean removeCaption) {
        return add("remove_caption", removeCaption);
    }

    public CopyMessages disableNotification(boolean disableNotification) {
        return add("disable_notification", disableNotification);
    }
    public CopyMessages protectContent(boolean protectContent) {
        return add("protect_content", protectContent);
    }

}

File: library/src/main/java/com/pengrad/telegrambot/request/CopyMessages.kt
package com.pengrad.telegrambot.request

import com.pengrad.telegrambot.response.MessageIdsResponse

class CopyMessages(
        chatId: Any,
        fromChatId: Any,
        messageIds: IntArray
) : BaseRequest<CopyMessages, MessageIdsResponse>(MessageIdsResponse::class.java) {
    init {
        add("chat_id", chatId).add("from_chat_id", fromChatId).add("message_ids", messageIds)
    }

    fun messageThreadId(messageThreadId: Int?): CopyMessages {
        return add("message_thread_id", messageThreadId)
    }

    fun removeCaption(removeCaption: Boolean): CopyMessages {
        return add("remove_caption", removeCaption)
    }

    fun disableNotification(disableNotification: Boolean): CopyMessages {
        return add("disable_notification", disableNotification)
    }

    fun protectContent(protectContent: Boolean): CopyMessages {
        return add("protect_content", protectContent)
    }
}
